desktop/api/src/main/java/org/datacleaner/windows/DataSetWindow.java

```diff
@@ -158,12 +170,12 @@ protected void done() {
                     _loadingIcon.setVisible(false);
                     _table.setVisible(true);
 
-                } catch (Exception e) {
+                } catch (Throwable e) {
+                    e = ErrorUtils.unwrapForPresentation(e);
+                    logger.error("Unexpected error occurred while building DataSetWindow contents", e);
                     DataSetWindow.this.dispose();
-                    if (e instanceof RuntimeException) {
-                        throw (RuntimeException) e;
-                    }
-                    throw new IllegalStateException(e);
+                    WidgetUtils.showErrorMessage("Unexpected error",
+                            "An unexpected error occurred while building data set. See logs for details.");
```

Comment:
This window is used to present data, but querying it or building the preview job is a pretty complex task that may quite likely go wrong. For this reason I felt it was better to hide the underlying exception from the user's view - The fact that it didn't work it most likely enough.